postgres/datadog_checks/postgres/metadata.py

```diff
@@ -275,98 +275,105 @@ def report_postgres_metadata(self):
+                            buffer_column_count = 0
+                            tables_buffer = []
 
-                                    tables_buffer = [*tables_buffer, *table_info]
-                                    for t in table_info:
-                                        buffer_column_count += len(t.get("columns", []))
+                            for tables in table_chunks:
+                                table_info = self._query_table_information(cursor, schema['name'], tables)
 
-                                    if buffer_column_count >= 100_000:
-                                        self._flush_schema(base_event, database, schema, tables_buffer)
-                                        total_tables += len(tables_buffer)
-                                        tables_buffer = []
-                                        buffer_column_count = 0
+                                tables_buffer = [*tables_buffer, *table_info]
+                                for t in table_info:
+                                    buffer_column_count += len(t.get("columns", []))
 
-                                if len(tables_buffer) > 0:
+                                if buffer_column_count >= 100_000:
                                     self._flush_schema(base_event, database, schema, tables_buffer)
                                     total_tables += len(tables_buffer)
-            except Exception as e:
-                self._log.error("Error collecting schema metadata: %s", e)
-                status = "error"
-            finally:
-                elapsed_ms = (time.time() - start_time) * 1000
-                self._check.histogram(
-                    "dd.postgres.schema.time",
-                    elapsed_ms,
-                    tags=self._check.tags + ["status:" + status],
-                    hostname=self._check.resolved_hostname,
-                    raw=True,
-                )
-                self._check.gauge(
-                    "dd.postgres.schema.tables_count",
-                    total_tables,
-                    tags=self._check.tags + ["status:" + status],
-                    hostname=self._check.resolved_hostname,
-                    raw=True,
-                )
-                datadog_agent.emit_agent_telemetry("postgres", "schema_tables_elapsed_ms", elapsed_ms, "gauge")
-                datadog_agent.emit_agent_telemetry("postgres", "schema_tables_count", total_tables, "gauge")
-
-                self._is_schemas_collection_in_progress = False
+                                    tables_buffer = []
+                                    buffer_column_count = 0
+
+                            if len(tables_buffer) > 0:
+                                self._flush_schema(base_event, database, schema, tables_buffer)
+                                total_tables += len(tables_buffer)
+        except Exception as e:
+            self._log.error("Error collecting schema metadata: %s", e)
+            status = "error"
```

Comment:
Shall we define the status as a constant?

Reply:
yes will do it separately. want to get this fix out first